### PR TITLE
git-credential-libsecret 2.33.0 (new formula)

### PR DESCRIPTION
**Pull request by FnControlOption**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/git-credential-libsecret.rb
Comment (SMillerDev) [inline Formula/git-credential-libsecret.rb:26]: We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

**Review by SMillerDev — COMMENTED** [review]
Why aren't we just building this as part of the default git install?

**Comment by FnControlOption** [comment]
> Why aren't we just building this as part of the default git install?

This requires `glib` and `libsecret` which aren't required by default git, but if it's acceptable, I can modify this pull request to edit the `git` formula.

**Comment by carlocab** [comment]
> This requires `glib` and `libsecret` which aren't required by default git, but if it's acceptable, I can modify this pull request to edit the `git` formula.

Yea, let's not do that. A separate formula is fine, I think.

File: Formula/git-credential-libsecret.rb
Comment (carlocab) [inline Formula/git-credential-libsecret.rb:4]: Let's add this comment to `git.rb` too. We can add it in a separate PR after this is merged to save CI time.

**Review by carlocab — APPROVED** [review]
LGTM. I'll merge this on Tuesday to wait for more feedback in case there are any objections.

**Review by SMillerDev — APPROVED** [review]

**Review by chenrui333 — APPROVED** [review]

**Pull request closed without merging**
